src/mod_private_mysql.erl

```diff
@@ -1,62 +0,0 @@
-%%% NS is namespace or key.
-%%% XML is #xmlel{} or value.
--module(mod_private_mysql).
--author('arcusfelis@gmail.com').
--behaviour(mod_private).
-
--export([init/2,
-         multi_set_data/3,
-         multi_get_data/3,
-         remove_user/2]).
-
--export([get_all_nss/2]).
-
--include("mongoose.hrl").
--include("jlib.hrl").
-
-init(_Host, _Opts) ->
-    ok.
-
-multi_set_data(LUser, LServer, NS2XML) ->
-    SLUser = mongoose_rdbms:escape_string(LUser),
-    Rows = [sql_row(NS, XML) || {NS, XML} <- NS2XML],
-    replace_like_insert_result(
-        rdbms_queries:multi_set_private_data(LServer, SLUser, Rows)).
-
-replace_like_insert_result({updated, _})        -> ok;
-replace_like_insert_result({error, Reason})     -> {error, Reason}.
-
-sql_row(NS, XML) ->
-    SNS = mongoose_rdbms:escape_string(NS),
-    SData = mongoose_rdbms:escape_string(exml:to_binary(XML)),
-    {SNS, SData}.
-
-multi_get_data(LUser, LServer, NS2Def) ->
-    SLUser = mongoose_rdbms:escape_string(LUser),
-    SNSs = [mongoose_rdbms:escape_string(NS) || {NS, _Def} <- NS2Def],
-    case rdbms_queries:multi_get_private_data(LServer, SLUser, SNSs) of
-        {selected, Rows} ->
-            RowsDict = dict:from_list(Rows),
-            [select_value(NSDef, RowsDict) || NSDef <- NS2Def];
-        _ ->
-            [Def || {_NS, Def} <- NS2Def]
-    end.
```

Comment:
Here we had one query, after the change there will be multiple queries - am I correct that the performance might go down? Do we agree to do that because the code is getting more consistent and readable?

Reply:
Most of private update requests are updating one namespace though. But the logic to optimize it later with a more generic solution.